mods/page_save.js

```diff
@@ -36,11 +36,49 @@ function PageSave(options) {
     };
 }
 
-PageSave.prototype._getRevInfo = function(restbase, req) {
+PageSave.prototype._getStartTimestamp = function(req) {
+    if (req.headers && req.headers['if-match']) {
+        var etag = rbUtil.parseETag(req.headers['if-match']);
+        if (!etag || !TimeUuid.test(etag.tid) || !/^(?:[0-9]+)$/.test(etag.rev)) {
+            throw new rbUtil.HTTPError({
```

Comment:
I'm ambivalent here. On the one hand, it's good to inform the client their header is not correct. On the other, we default to using the current *ts*.